engine/apps/slack/scenarios/distribute_alerts.py

```diff
@@ -50,68 +46,92 @@
 logger.setLevel(logging.DEBUG)
 
 
-class AlertShootingStep(scenario_step.ScenarioStep):
+class IncomingAlertStep(scenario_step.ScenarioStep):
     def process_signal(self, alert: Alert) -> None:
```

Comment:
the logic in `IncomingAlertStep.process_signal` _should be_ functionally the same. I made a few cleanups (like assigning things to variables) and also think I got rid of a few unnecessary SQL queries.

For example, when we were doing `AlertGroup.objects.filter(pk=alert.group.pk, ...`, we don't need to do an additional `SELECT` query, since we already have this `AlertGroup` available via `alert.group` (`alert` is passed as the sole argument to `IncomingAlertStep.process_signal`).

See the `_mark_alert_group_slack_message_as_sent` function for more context (defined as a nested function inside of `process_signal`)